openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -13,272 +13,263 @@
-        "image_url": img_url,
-        "license_info": get_license_info(license_url=license_url),
-        "width": width,
-        "height": height,
-        "creator": creator,
-        "title": title,
-        "meta_data": meta_data,
-    }
-    return details
-
-
-def _get_creator_details(result):
-    credit_line = None
-    pub_date = None
-    creator = None
-    first_name = result.get("submitter", {}).get("firstName")
-    last_name = result.get("submitter", {}).get("lastName")
-    if first_name and last_name:
-        creator = f"{first_name} {last_name}".strip()
-
-    if result.get("credit"):
-        credit_line = result.get("credit").strip()
-        pub_date = result.get("submitted").strip()
-
-    return creator, credit_line, pub_date
-
-
-def _get_taxa_details(result):
-    taxa = result.get("taxa", [])
-    # [0].get('canonicalName', {}).get('string')
-    taxa_list = None
-    title = ""
-    if taxa:
-        taxa = [
-            _.get("canonicalName") for _ in taxa if _.get("canonicalName") is not None
-        ]
-        taxa_list = [_.get("string", "") for _ in taxa]
-
-    if taxa_list:
-        title = taxa_list[0]
-
-    return taxa_list, title
-
-
-def _get_image_info(result, _uuid):
-    base_url = "http://phylopic.org"
-    img_url = None
-    width = None
-    height = None
-
-    image_info = result.get("pngFiles")
-    if image_info:
-        images = list(
-            filter(lambda x: (int(str(x.get("width", "0"))) >= 257), image_info)
-        )
-        if len(images) > 0:
-            image = sorted(images, key=lambda x: x["width"], reverse=True)[0]
-            img_url = image.get("url")
-            if not img_url:
-                logging.warning(f"Image not detected in url: {base_url}/image/{_uuid}")
-            else:
-                img_url = f"{base_url}{img_url}"
-                width = image.get("width")
-                height = image.get("height")
-
-    return img_url, width, height
-
-
-def _compute_date_range(date_start: str, days: int = DEFAULT_PROCESS_DAYS) -> str:
-    """
-    Given an ISO formatted date string and a number of days, compute the
-    ISO string that represents the start date plus the days provided.
-    """
-    date_end = date.fromisoformat(date_start) + timedelta(days=days)
-    return date_end.isoformat()
+class PhylopicDataIngester(ProviderDataIngester):
+    delay = 5
+    host = "http://phylopic.org"
+    # Use "base_endpoint" since class's "endpoint" parameter gets defined as a property
+    base_endpoint = f"{host}/api/a/image"
+    providers = {"image": prov.PHYLOPIC_DEFAULT_PROVIDER}
+    batch_limit = 25
+    # If a date is supplied to the ingester in the form of YYYY-MM-DD, this is the
+    # number of days from that date to process. By default,this DAG runs weekly
+    # and so default_process_days is 7.
+    default_process_days = 7
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
Can we make Phylopic accept `process_days` as an argument here instead of overriding the `default` when we want to change it (as is done in the `main` method)?

This will be necessary for the reingestion workflows, where we want to be able to run Phylopic for individual days instead of a week at a time. Even with that change, this still introduces a complication for reingestion since those workflows are factory-generated. We would need to add an `args` option to the ProviderReingestionWorkflow configuration objects or something similar, that we could pass in through the factory 😬 

Outside of the scope of this PR, I'd like to at least consider making this a daily DAG. If it is giving us a small enough amount of data that this is unreasonable, then alternatively maybe it shouldn't be dated?

Reply:
I tried doing this initially, and I ran into some issue with handling `args` vs `kwargs` vs this parameter in the various places that the provider ingester class is passed around. I think having it as `@weekly` was fine initially, but if that's going to create more problems for the reingestion workflow, then maybe this would just be better as `@daily` 😄 It'd prevent us from having to figure out all these additional workarounds for `process_days` as well. I'll go ahead and make that change as a commit, but if we do want to pursue just keeping this as a weekly DAG before merging I can revert that commit.